### PR TITLE
feat(resource-util): add support to detect Cloud Run and Cloud Functions

### PR DESCRIPTION
**Pull request by aabmass**
Copied from the Go impl.

E2E tests are also passing in follow up PR https://github.com/GoogleCloudPlatform/opentelemetry-operations-js/pull/480

### CODE REVIEW

File: packages/opentelemetry-resource-util/src/index.ts [hunk 1]
```diff
@@ -110,6 +116,16 @@ const MAPPINGS = {
...
   },
+  [CLOUD_RUN_REVISION]: {
+    [LOCATION]: {otelKeys: [SemanticResourceAttributes.CLOUD_REGION]},
+    [SERVICE_NAME]: {otelKeys: [SemanticResourceAttributes.FAAS_NAME]},
+    [CONFIGURATION_NAME]: {otelKeys: [SemanticResourceAttributes.FAAS_NAME]},
+    [REVISION_NAME]: {otelKeys: [SemanticResourceAttributes.FAAS_VERSION]},
+  },
+  [CLOUD_FUNCTION]: {
+    [REGION]: {otelKeys: [SemanticResourceAttributes.CLOUD_REGION]},
+    [FUNCTION_NAME]: {otelKeys: [SemanticResourceAttributes.FAAS_NAME]},
```
Comment (aabmass) [inline packages/opentelemetry-resource-util/src/index.ts:127]: These mappings are used for both trace `g.co/r/` labels and monitoring monitored resource. These monitored resources are not currently writable for monitoring, so this presents an issue (we have the same issue in the java impl I believe https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/blob/60cf3d675eda0900258853d568c21bfd4d692d49/shared/resourcemapping/src/main/java/com/google/cloud/opentelemetry/resource/ResourceTranslator.java#L97

Should we go ahead with this or maybe I can add a `skipUnsupportedMonitoredResources` bool option?

Comment (dashpole) [reply packages/opentelemetry-resource-util/src/index.ts:127]: You can see what I did in go: https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/main/exporter/metric/metric.go#L343

Either option is fine with me

Comment (aabmass) [reply packages/opentelemetry-resource-util/src/index.ts:127]: Disabled auto-merge. I will update it so it works correctly on cloud run

Comment (aabmass) [reply packages/opentelemetry-resource-util/src/index.ts:127]: Updated L187 ptal

**Review by dashpole — APPROVED** [review]

File: packages/opentelemetry-resource-util/src/index.ts (hunk 1 above)
Comment (mbrevda) [reply packages/opentelemetry-resource-util/src/index.ts:127]: appologies for waking a dead thread: could you possible explain "These monitored resources are not currently writable for monitoring" and/or point to reference documentation?

**Pull request merged**